test/test-fork.c

```diff
@@ -346,6 +346,33 @@ TEST_IMPL(fork_signal_to_child_closed) {
   return 0;
 }
 
+static void empty_signal_cb(uv_signal_t* h, int s){}
+static void empty_close_cb(uv_handle_t* h){}
+
+TEST_IMPL(fork_close_signal_in_child) {
+  uv_loop_t loop;
+  uv_signal_t signal_handle;
+  pid_t child_pid;
+
+  uv_loop_init(&loop);
+  uv_signal_init(&loop, &signal_handle);
+  uv_signal_start(&signal_handle, &empty_signal_cb, SIGHUP);
+
+  kill(getpid(), SIGHUP);
+  child_pid = fork();
```

Comment:
Can you check that the signal callback:

1. Has not been called at this point, right after fork(), but
2. Has been called in the parent process when `uv_run()` returns, but not in the child process?

Aside: please also close the signal handle in the parent process, followed by a call to `uv_run(UV_RUN_DEFAULT)`.

Reply:
Thank you for review. I fixed everything in the next commit.